contracts/SpokePoolV3Periphery.sol

```diff
@@ -581,30 +686,50 @@ contract SpokePoolV3Periphery is Lockable, MultiCaller {
 
     // This contract supports two variants of swap and bridge, one that allows one token and another that allows the caller to pass them in.
     function _swapAndBridge(
-        address exchange,
-        bytes calldata routerCalldata,
-        uint256 swapTokenAmount,
-        uint256 minExpectedInputTokenAmount,
+        SwapData calldata swapData,
         DepositData calldata depositData,
-        IERC20 _swapToken,
         IERC20 _acrossInputToken
     ) private {
+        // Load variables we use multiple times onto the stack.
+        IERC20 _swapToken = swapData.swapToken;
+        TransferType _transferType = swapData.transferType;
+        address _exchange = swapData.exchange;
+        uint256 _swapTokenAmount = swapData.swapTokenAmount;
         // Swap and run safety checks.
         uint256 srcBalanceBefore = _swapToken.balanceOf(address(this));
         uint256 dstBalanceBefore = _acrossInputToken.balanceOf(address(this));
 
-        _swapToken.forceApprove(exchange, swapTokenAmount);
+        if (_transferType == TransferType.Approval) _swapToken.forceApprove(_exchange, _swapTokenAmount);
+        else if (_transferType == TransferType.Transfer) _swapToken.transfer(_exchange, _swapTokenAmount);
+        else {
+            permit2.permit(
+                address(this), // owner
+                IPermit2.PermitSingle({
```

Comment:
Seems very strange, but we should also just be able to make these all just dummy values, since we never read anything from permit2, which is the contract that uses these values to process the approval. Sounds odd though.